File: fixtures/debuggee/fizz_engine.py
#!/usr/bin/env python3

import os
from pathlib import Path


def main():
    Path(__file__).with_name('debuggee.pid').write_text(str(os.getpid()), encoding='ascii')
    engine = build_engine()
    values = [int(x) for x in engine['items'] if x.isdigit()]
    total = sum(values)
    report = {'total': total, 'count': len(engine['items'])}
    print(report)


def build_engine():
    items = [
        '1', '2', 'Fizz', '4', 'Buzz',
        'Fizz', '7', '8', 'Fizz', 'Buzz',
        '11', '10'
    ]
    return {'items': items}


if __name__ == '__main__':
    main()

File: fixtures/debuggee/fizz_engine.js
const fs = require('fs');

function buildItems(limit) {
  const items = [];
  for (let i = 1; i <= limit; i++) {
    if (i % 15 === 0) items.push('FizzBuzz');
    else if (i % 3 === 0) items.push('Fizz');
    else if (i % 5 === 0) items.push('Buzz');
    else items.push(String(i));
  }
  return items;
}

function main() {
  const engine = {
    name: 'fizz-engine',
    items: buildItems(15),
    nested: { meta: { depth: 3, active: true } },
  };
  const total = engine.items.reduce((sum, item) => sum + String(item).length, 0);
  const report = { ok: true, total, engine };
  const pidFile = path.join(__dirname, 'debuggee.pid');
  fs.writeFileSync(pidFile, String(process.pid));
  console.log(JSON.stringify(report));
}

main();
